osbs/constants.py

```diff
@@ -79,6 +79,10 @@
 REPO_CONFIG_FILE = '.osbs-repo-config'
 ADDITIONAL_TAGS_FILE = 'additional-tags'
 REPO_CONTAINER_CONFIG = 'container.yaml'
+REPO_CONTAINER_CONFIG_WRONG_EXTENSION = 'container.yml'
+REPO_CONTENT_SETS_FILE = 'content_sets.yml'
+REPO_CONTENT_SETS_FILE_WRONG_EXTENSION = 'content_sets.yaml'
```

Comment:
I have crazy idea what about having sets of potential typos?

```python
REPO_CONTAINER_CONFIG_WRONG = {'container.yml', 'contianer.yaml'}
REPO_CONTENT_SETS_FILE_WRONG = {'content_sets.yaml', 'content-sets.yml'}
```

Reply:
I have another crazy idea after reading this PR which makes me remind of another task that validates a few user config files inside an atomic-reactor plugin with predefined JSON schema.

To be short, we could abstract a container repository into a dedicated object.

The object could provides methods to check and validate user config files with JSON schema and properties to get the file names or read file content. For example:

```python
class ContainerRepo:
    def __init__(self, root_dir):
        self._root_dir = root_dir
        # check files, e.g. ensure the correct container.yaml file name

    def validate(self):
        # validate various user configs, e.g. content_set.yml
        # there could be separate validation method for each specific user config file

    @property
    def dockerfile(self):
        return os.path.join(self._root_dir, 'Dockerfile')

    @property
    def content_set_yml(self):
        return os.path.join(self._root_dir, 'content_sets.yml')

    @property
    def container_yml(self):
        return os.path.join(self._root_dir, 'container.yml')

    def read_content_sets(self):
        """Return content of the content_sets.yml"""
```

With this abstraction, all the code related to handling the files inside a container repo will be in one place, not spread into osbs-client and atomic-reactor, and we could handle more things inside this object as well.

Another benefit would be by reading this object, it should be much easier to know what files could be included in a container repo and what files should be included for a successful container build.

I'm not requesting changes to this PR based on this thought. Just sharing my thought and FYI.

Reply:
@tkdchen feel free to file a story, mbasti said he likes the idea too.